AndroidSDKFcm/src/main/java/com/leanplum/LeanplumPushFirebaseMessagingService.java

```diff
@@ -86,4 +99,63 @@ private Bundle getBundle(Map<String, String> messageMap) {
     }
     return bundle;
   }
+
+  private Boolean isDuplicateNotification(Map<String, String> messageMap) {
+    Boolean isDuplicate = false;
+    String messageId = "";
+    if (messageMap.containsKey("_lpn")) {
```

Comment:
This needs checking. Please reference the existing duplicate resolution code. Looks like we might be using the backend provided message Id. Also, will go through this code in person, little confused here.